trezorlib/client.py

```diff
@@ -1100,6 +1101,172 @@ def self_test(self):
 
         return self.call(proto.SelfTest(payload=b'\x00\xFF\x55\xAA\x66\x99\x33\xCCABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!\x00\xFF\x55\xAA\x66\x99\x33\xCC'))
 
+    @expect(proto.StellarPublicKey)
+    def stellar_get_public_key(self, address_n):
+        return self.call(proto.StellarGetPublicKey(address_n=address_n))
+
+    def stellar_sign_transaction(self, txEnvelope, address_n, networkPassphrase=None):
+        # default networkPassphrase to the public network
+        if networkPassphrase is None:
+            networkPassphrase = "Public Global Stellar Network ; September 2015"
+
+        parsed = stellar.parse_transaction_bytes(txEnvelope)
+
+        # Will return a StellarTxOpRequest
+        resp = self.call(proto.StellarSignTx(
+            protocol_version=parsed["protocol_version"],
+            address_n=address_n,
+            network_passphrase=networkPassphrase,
+            source_account=parsed["source_account"],
+            fee=parsed["fee"],
+            sequence_number=parsed["sequence_number"],
+            timebounds_start=parsed["timebounds_start"],
+            timebounds_end=parsed["timebounds_end"],
+            memo_type=parsed["memo_type"],
+            memo_text=parsed["memo_text"],
+            memo_id=parsed["memo_id"],
+            memo_hash=parsed["memo_hash"],
+            num_operations=parsed["num_operations"]
+        ))
+        if resp.__class__.__name__ != "StellarTxOpRequest":
+            raise CallException("Unexpected response to transaction")
+
+        for opIdx in range(0, parsed["num_operations"]):
+            op = parsed["operations"][opIdx]
+            resp = None
+
+            # Create account
+            if op["type"] == 0:
```

Comment:
this whole series of checks should be its own function(s) in `stellar.py`

also maybe use constants for the types? I imagine something like
```python
OP_CREATE_ACCOUNT = 0
OP_PAYMENT = 1
# ...
def make_operation(op):
    if op["type"] == OP_CREATE_ACCOUNT:
       return proto.StellarCreateAccountOp(...)
    if op["type"] == OP_PAYMENT:
       return proto.StellarPaymentOp(...)
    # ....
```
then in client.py just:
```python
for op in parsed["operations"]:
   resp = self.call(stellar.make_operation(op))
   if isinstance(resp, StellarSignedTx):
        return resp
```